Add documentation for Event Handler chart (#13732)
gravitational/teleport: docs/pages/setup/guides/fluentd.mdx

  $ tar -zxvf teleport-event-handler-v(=teleport.plugin.version=)-linux-amd64-bin.tar.gz
  ```
  </TabItem>

  <TabItem label="Docker">
  ```code
  $ docker pull quay.io/gravitational/teleport-plugin-event-handler:(=teleport.plugin.version=)
  ```
  </TabItem>

  <TabItem label="Helm">
  ```code
  $ helm repo add teleport https://charts.releases.teleport.dev/
  ```

>   See the Helm Chart example under [Step 6/6. Start the event handler plugin](#step-66-start-the-event-handler-plugin) to see how to create `teleport-plugin-event-handler-values.yaml`!

Reviewer: Would a Helm user still need to create a Teleport user/role for reading audit events, enable impersonation for the `teleport-event-handler` user/role, and export credentials? If so, I think we should delete this sentence and make sure each of steps 2-6 includes Helm instructions.

For example, we can rename Step 2 to "Configure the plugin", remove the instructions below the `Tabs` box, and copy those instructions into the `TabItem`s for "Self-Hosted" and "Teleport Cloud". We could then add a third tab for "Helm" that provides instructions on creating a values file. 

Currently, the values file instructions are in Step 6, which is why we need to jump around the doc.

  $ tar -zxvf teleport-event-handler-v(=teleport.plugin.version=)-linux-amd64-bin.tar.gz
  ```
  </TabItem>

  <TabItem label="Docker">
  ```code
  $ docker pull quay.io/gravitational/teleport-plugin-event-handler:(=teleport.plugin.version=)
  ```
  </TabItem>

  <TabItem label="Helm">
  ```code
  $ helm repo add teleport https://charts.releases.teleport.dev/

>   $ helm install teleport-plugin-event-handler teleport/teleport-plugin-event-handler --values teleport-plugin-event-handler-values.yaml

Reviewer: So the reader doesn't need to jump around the page, maybe we could limit this snippet to the `helm repo add` command, then explain that we will configure the plugin and run it later in the guide. We won't need to link to Step 6, since the reader will eventually get there.

I would recommend incorporating the second half of Step 6 (running the `teleport-event-handler start` command) into each of the "Teleport Cloud" and "Self-Hosted" `TabItem`s, then adding an example command to the "Helm Chart" `TabItem` in Step 6 to install the chart with the given values file.

Author: I moved it there now, but I feel like it's confusing if the other command (starting it directly) is immediately under the tab view. How about moving the command into the other tabs as well (Teleport Cloud and Self-hosted)?

$ ./teleport-event-handler configure . mytenant.teleport.sh
```

</TabItem>
<TabItem scope={["oss", "enterprise", "cloud"]} label="Helm Chart">

> Run the `configure` command to generate a sample configuration. Replace `mytenant.teleport.sh` with the DNS name of your Teleport server (Teleport Cloud or self-hosted):

Reviewer: ```suggestion
Run the `configure` command to generate a sample configuration. Assign `TELEPORT_CLUSTER_ADDRESS` to the DNS name and port of your Teleport Auth Service or Proxy Service:
```
Suggesting using an env variable since `mytenant.teleport.sh` is not visible unless you scroll

> ```code

Reviewer: ````suggestion
```code
$ TELEPORT_CLUSTER_ADDRESS=mytenant.teleport.sh:443
````

> $ docker run -v `pwd`:/opt/teleport-plugin -w /opt/teleport-plugin quay.io/gravitational/teleport-plugin-event-handler:(=teleport.plugin.version=) configure . mytenant.teleport.sh

Reviewer: ```suggestion
$ docker run -v `pwd`:/opt/teleport-plugin -w /opt/teleport-plugin quay.io/gravitational/teleport-plugin-event-handler:(=teleport.plugin.version=) configure . ${TELEPORT_CLUSTER_ADDRESS?}
```
So it's slightly easier to copy and paste the command

$ tctl auth sign --user=teleport-event-handler --out=identity
```

> The above sequence should result in one PEM-encoded file, `terraform-identity`.

Reviewer: ```suggestion
The above sequence should result in one PEM-encoded file, `identity`.
```

The above sequence should result in one PEM-encoded file: `terraform-identity`.

</TabItem>

> <TabItem label="Helm chart" scope={["cloud"]}>

Reviewer: ```suggestion
<TabItem label="Helm Chart" scope={["cloud"]}>
```
To be consistent with the other tab labels

$ tctl auth sign --format=tls --user=teleport-event-handler --out=auth
```

> This command should result in three PEM-encoded files, `auth.crt`, `auth.key`,

Reviewer: ```suggestion
This command should result in three PEM-encoded files: `auth.crt`, `auth.key`,
```
I think the colon is clearer since it signals that a list will follow

$ tctl auth sign --user=teleport-event-handler --out=identity
```

The above sequence should result in one PEM-encoded file, `identity`.

</TabItem>
<TabItem label="Helm Chart" scope={["cloud"]}>

If you are planning to use the Helm Chart, you'll need to generate the keys
with the `file` format, then create a secret in Kubernetes.

Create the identity using the following command:

```code
$ tctl auth sign --format=file --user=teleport-event-handler --out=identity
```

> Then create the Kubernetes secret:

Reviewer: Is there a way to prevent duplicating the `kubectl create secret` command between this tab and the "Configure the plugin" tab above? Maybe remove the command in the "Configure the plugin" tab since we need the `tctl auth sign` command to generate the credentials used in the `kubectl create secret` command?

Author: The "Configure the plugin" one is creating a different secret for the TLS credentials generated by `teleport-plugin-event-handler configure`. Or do you mean we should put them into one secret? I'd rather keep them separate to indicate that one's used for connecting to Teleport, the other is used with Fluentd.

Reviewer: Oh, I see now. One issue is that in the "Configure the plugin" section, we mention the `kubectl create secret` command before we explain that the `configure` command creates the TLS credential files.

We could add a quick sentence before "To create a Kubernetes secret" explaining why we're creating the secret and how we got the files we're referring to in the `kubectl create secret` command.

Author: Good point, thanks!

Author: I rephrased the first secret creation, what do you think?

root_cas = "auth.cas"
```

</TabItem>
<TabItem scope={["oss", "enterprise"]} label="Helm chart">

Use the following template to create values file:

```yaml
eventHandler:
  storagePath: "./storage"
  timeout: "10s"
  batch: 20
  namespace: "default"

fluentd:
  url: "https://localhost:8888/test.log"

>   caPath: "/home/sasha/scripts/event-handler/ca.crt"

Reviewer 2: Are these values correct? Should it be the path inside the secret or does this secret get mounted?
